plugins/frontend/backstage-plugin-prometheus/src/hooks/usePrometheus.ts

```diff
@@ -132,10 +132,23 @@ export function useAlerts(alerts: string[] | 'all') {
     }, []);
   if (value && value.status !== 'error') {
     const rules = value.data.groups.flatMap(it => it.rules);
+    const labels = getLabels(entity);
+    const parsedLabels =
+      labels &&
+      Object.fromEntries(labels.split(',').map(label => label.split(/=(.*)/s)));
     const displayableAlerts: PrometheusDisplayableAlert[] = rules
       .filter(rule => alerts === 'all' || alerts.includes(rule.name))
       .flatMap(rule =>
         rule.alerts.map(alert => ({ ...rule, ...alert, id: rule.name })),
+      )
+      .filter(
+        alert =>
+          parsedLabels &&
```

Comment:
I guess, this check is in here in case the `parsedLabels` are `undefined` correct? What this filter does when there is no label selector?